docs/local-development-guide.md

```diff
@@ -0,0 +1,192 @@
+- Gradle 8.5+ (for building Java-based services)
+
+## Building the Coordinator Locally
+
+The coordinator is a Java-based service that orchestrates the Linea protocol's operations. You can build it locally using the following steps:
+
+### 1. Clone the Repository
+
+If you haven't already, clone the repository and navigate to the project directory:
+
+```bash
+git clone https://github.com/ConsenSys/linea-monorepo.git
+cd linea-monorepo
+```
+
+### 2. Install Dependencies
+
+Install the Node.js dependencies:
+
+```bash
+make pnpm-install
+```
+
+### 3. Build the Coordinator
+
+The coordinator can be built using Gradle:
+
+```bash
+./gradlew :coordinator:build
+```
+
+This will generate the coordinator JAR file in the `coordinator/app/build/libs` directory.
+
+### 4. Building the Coordinator Docker Image Locally
+
+To build a local Docker image of the coordinator:
+
+```bash
+# First build the coordinator JAR
+./gradlew :coordinator:build
+
+# Then build the Docker image
+docker build -t linea-coordinator-local -f coordinator/Dockerfile --build-arg VERSION=$(git rev-parse --short HEAD) .
+```
+
+## Running the Coordinator Locally
+
+There are two main ways to run the coordinator:
+
+### 1. Running as Part of the Full Stack
+
+The recommended way to run the coordinator is as part of the complete Linea stack:
+
+```bash
+# Start the entire stack with tracing v2
+make start-env-with-tracing-v2
+```
+
+This command:
+1. Starts all necessary services (L1 node, L2 node, sequencer, etc.)
+2. Deploys the required smart contracts
+3. Starts the coordinator service
+
+### 2. Running the Coordinator Standalone
+
+If you want to run just the coordinator for development purposes, you can use:
+
+```bash
+# Set necessary environment variables
+export POSTGRES_USER=postgres
+export POSTGRES_PASSWORD=postgres
+
+# Run only PostgreSQL database (required by coordinator)
+docker compose -f docker/compose-tracing-v2.yml up -d postgres
+
+# Run the coordinator with your local build
+java -Dvertx.configurationFile=config/coordinator/vertx-options.json \
+     -Dlog4j2.configurationFile=config/coordinator/log4j2-dev.xml \
+     -jar coordinator/app/build/libs/coordinator.jar \
+     --traces-limits-v2 config/common/traces-limits-v2.toml \
+     --smart-contract-errors config/common/smart-contract-errors.toml \
+     --gas-price-cap-time-of-day-multipliers config/common/gas-price-cap-time-of-day-multipliers.toml \
+     config/coordinator/coordinator-docker.config.toml \
+     config/coordinator/coordinator-docker-traces-v2-override.config.toml
+```
+
+Note: When running the coordinator standalone, you'll need to ensure that all its dependencies (such as L1 and L2 nodes) are properly configured and running.
+
+## Using Your Locally Built Coordinator Image
+
+To use your locally built coordinator image with the existing setup:
+
+1. Build the local image as shown above
+2. Edit the `docker/compose-spec-l2-services.yml` file to use your local image:
```

Comment:
If we follow the suggested steps above, we don't need to manually edit the `docker/compose-spec-l2-services.yml` file at all, instead we can just skip this step and in step 3 do:
```
COORDINATOR_TAG=local make start-env-with-tracing-v2
```